Feature/when next upgrades

JR-Mitchell opened the pull request:
### What?

- Removes the argument parsing logic from `when_next.py`
- Adds new method arguments to `when_next.py` `entry()` method

### Why?

The argument parsing didn't work, due to space in commands delimiting new method parameters.
Therefore, this logic has been replaced with positional arguments.

JR-Mitchell committed:
Removes non-working argument parsing logic from when_next.py

JR-Mitchell committed:
Adds multiple arguments to the when_next.entry method to provide greater flexibility

JR-Mitchell committed:
Fixes linting issues

JR-Mitchell committed:
Fixes line spaces

JR-Mitchell committed:
Ensures consistent parameter casing

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
+                context.channel,
+                discord.Embed(
+                    title="Hang-out planning",
+                    description="Uhhhh so when do you wanna hang out?",
```
I think this needs one more `h`

JR-Mitchell replied:
Resolved as of 1e786a9f

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
-                return
-        weekend_days = []
+        days_int = self._parse_int(context, "days", days)
+        if days_int is False:
```
I don't know if you used `is` over `==` deliberately, but despite being a syntax warning, this saves you if `days_int == 0` lol

JR-Mitchell replied:
I think I just googled which way you should check falsiness in python and used that

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
-            if until_date is False:
-                return
-        weekend_days = []
+        days_int = self._parse_int(context, "days", days)
```
Needs to be awaited.

JR-Mitchell replied:
Resolved as of 1e786a9f

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
+        days_int = self._parse_int(context, "days", days)
+        if days_int is False:
+            return
+        offset_int = self._parse_int(context, "start_offset", start_offset)
```
Await

JR-Mitchell replied:
Resolved as of 1e786a9f

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
         while from_date < until_date:
-            if from_date.weekday() > 4:
-                weekend_days.append(from_date.strftime(DATE_FORMAT))
+            if str(from_date.weekday()) in weekdays:
```
Excellent hack

JR-Mitchell replied:
ty

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
+                timeout=1,
+            )
+        else:
+            await context.send(f"Invalid combination of parameters led to no valid days to vote on")
```
Can you add a bit more logging so it's possible to work out if something blew up when putting `valid_days` together? Just an `info` or maybe even `debug` before the `for` loop that logs the `from_date` and `until_date` would be useful.

JR-Mitchell replied:
Logging added as of 1e786a9f

fjebaker reviewed (changes requested):
Jawohl, gut so 👍

fjebaker commented:
@JR-Mitchell have you had a chance to test this on a dev bot or is this a lucky dip?

fjebaker commented:
I've been meaning to setup webhooks to get the server to automatically pull and relaunch when pushes to `main` are made, so might do that over the weekend. Should hopefully improve time-to-fail.

JR-Mitchell committed:
Makes changes to resolve PR conversation
- Awaits calls to _parse_int()
- Logs info relevant to calculating dates
- Adds additional 'h' character

JR-Mitchell commented:
> @JR-Mitchell have you had a chance to test this on a dev bot or is this a lucky dip?

@fjebaker lucky dip - I don't have an environment for testing set up

fjebaker commented on src/cogs/when_next.py:
```diff
@@ -48,7 +49,7 @@ async def entry(self, context, days: str = "31", weekdays: str = "567", start_of
                 discord.Embed(
                     title="Hang-out planning",
-                    description="Uhhhh so when do you wanna hang out?",
+                    description="Uhhhhh so when do you wanna hang out?",
```
👍

fjebaker reviewed (approved):
LGTM
